maap/config_reader.py

```diff
@@ -1,278 +1,305 @@
-        self.__s3_user_upload_bucket = os.environ.get("S3_USER_UPLOAD_BUCKET") or self.__config.get("aws",
-                                                                                                  "user_upload_bucket")
-        self.__s3_user_upload_dir = os.environ.get("S3_USER_UPLOAD_DIR") or self.__config.get("aws",
-                                                                                            "user_upload_directory")
-        self.__indexed_attributes = json.loads(self.__config.get("search", "indexed_attributes"))
+            raise EnvironmentError("MAAP is not configured. Please initialize a MAAP object: maap = MAAP()")
+        self.__config = _get_client_config(maap_host)
+        self.maap_host = maap_host
+        self.maap_api_root = self.__config.get("service").get("maap_api_root")
+        self.maap_token = self.__config.get("service").get("maap_token")
+        self.page_size = os.environ.get("MAAP_CMR_PAGE_SIZE", 20)
+        self._PROXY_GRANTING_TICKET = os.environ.get("MAAP_PGT", '')
+        self.content_type = os.environ.get("MAAP_CMR_CONTENT_TYPE", "application/echo10+xml")
+        self.algorithm_register = self._get_api_endpoint("algorithm_register")
+        self.algorithm_build = self._get_api_endpoint("algorithm_build")
+        self.mas_algo = self._get_api_endpoint("mas_algo")
+        self.dps_job = self._get_api_endpoint("dps_job")
+        self.member_dps_token = self._get_api_endpoint("member_dps_token")
+        self.requester_pays = self._get_api_endpoint("requester_pays")
+        self.edc_credentials = self._get_api_endpoint("edc_credentials")
+        self.workspace_bucket_credentials = self._get_api_endpoint("workspace_bucket_credentials")
+        self.s3_signed_url = self._get_api_endpoint("s3_signed_url")
+        self.wmts = self._get_api_endpoint("wmts")
+        self.member = self._get_api_endpoint("member")
+        self.tiler_endpoint = self.__config.get("service").get("tiler_endpoint")
+        self.aws_access_key = os.environ.get("MAAP_AWS_ACCESS_KEY_ID")
+        self.aws_access_secret = os.environ.get("MAAP_AWS_SECRET_ACCESS_KEY")
+        self.s3_user_upload_bucket = os.environ.get("MAAP_S3_USER_UPLOAD_BUCKET")
+        self.s3_user_upload_dir = os.environ.get("MAAP_S3_USER_UPLOAD_DIR")
+        self.search_granule_url = self._get_api_endpoint("search_granule_url")
+        self.search_collection_url = self._get_api_endpoint("search_collection_url")
+        self.indexed_attributes = self.__config.get("search").get("indexed_attributes")
+        self.mapbox_token = os.environ.get("MAAP_MAPBOX_ACCESS_TOKEN", '')
 
     def _get_api_endpoint(self, config_key):
-        return 'https://{}/api/{}'.format(self.maap_host, self.__config.get("maap_endpoint", config_key))
+        endpoint = str(self.__config.get("maap_endpoint").get(config_key)).strip("/")
+        return os.path.join(self.maap_api_root, endpoint)
```

Comment:
Use `urljoin`, not `path.join` since you're creating a URL, not a path on a filesystem. URLs are not OS-dependent.

```suggestion
        return urllib.parse.urljoin(self.maap_api_root, endpoint)
```